test/unused-fields.js

```diff
@@ -109,6 +123,26 @@ ruleTester.run('unused-fields', rules['unused-fields'], {
       `,
       errors: [unusedFieldsWarning('unused1'), unusedFieldsWarning('unused2')]
     },
+    {
+      code: `
+        graphql\`fragment Test on Page { unused1 { unused2 } }\`;
+      `,
+      errors: [unusedFieldsWarning('unused1'), unusedFieldsWarning('unused2')]
+    },
+    {
+      code: `
+        graphql\`fragment Test on Page { unused1 { unused2 } }\`;
+      `,
+      options: [{leavesOnly: true}],
+      errors: [unusedFieldsWarning('unused2')]
+    },
```

Comment:
does this works well if we use options, for some reason, as `[{leavesOnly: true}, ignoreFields: ['unsed2']]`?
I think that can add a new test
```
{
  code: `
    graphql\`fragment Test on Page { unused1 { unused2 } }\`;
  `,
  options: [{leavesOnly: true}, ignoreFields: ['unsed2']],
},
```